Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift

```diff
@@ -307,24 +310,29 @@ extension BrowserViewController: TopToolbarDelegate {
       // Do not allow users to enter URLs with the following schemes.
       // Instead, submit them to the search engine like Chrome-iOS does.
       if !["file"].contains(fixupURL.scheme) {
-        // check text is SNS domain
+        // check text is decentralized DNS supported domain
         let isPrivateMode = PrivateBrowsingManager.shared.isPrivateBrowsing
-        if !isPrivateMode, fixupURL.domainURL.schemelessAbsoluteDisplayString.endsWithSupportedSNSExtension, let rpcService = BraveWallet.JsonRpcServiceFactory.get(privateMode: isPrivateMode) {
-          let currentStatus = await rpcService.snsResolveMethod()
-          switch currentStatus {
-          case .ask:
-            // show name service interstitial page
-            showSNSDomainInterstitialPage(originalURL: fixupURL, visitType: visitType)
+        if DecentralizedDNSHelper.isSupported(domain: fixupURL.domainURL.schemelessAbsoluteDisplayString),
+           let rpcService = BraveWallet.JsonRpcServiceFactory.get(privateMode: isPrivateMode) {
+          let decentralizedDNSHelper = DecentralizedDNSHelper(rpcService: rpcService, ipfsApi: braveCore.ipfsAPI)
+          topToolbar.leaveOverlayMode()
+          updateToolbarCurrentURL(fixupURL)
+          topToolbar.locationView.loading = true
+          let result = await decentralizedDNSHelper.lookup(domain: fixupURL.schemelessAbsoluteDisplayString)
+          topToolbar.locationView.loading = tabManager.selectedTab?.loading ?? false
+          guard !Task.isCancelled else { return true } // user pressed stop, or typed new url
+          switch result {
+          case let .loadInterstitial(service):
+            showWeb3ServiceInterstitialPage(service: service, originalURL: fixupURL, visitType: visitType)
             return true
-          case .enabled:
-            if let resolvedURL = await resolveSNSHost(text, rpcService: rpcService) {
-              // resolved url
+          case let .load(resolvedURL):
+            if resolvedURL.isIPFSScheme {
+              handleIPFSSchemeURL(resolvedURL, visitType: visitType)
+            } else {
               finishEditingAndSubmit(resolvedURL, visitType: visitType)
-              return true
```

Comment:
why are we removing the `return` here?

Reply:
We're not, the return is right below this line? But looks like I lost the `return` for `handleIPFSSchemeURL(resolvedURL, visitType: visitType)` on a rebase.

Reply:
Updated to use return value from `handleIPFSSchemeURL` function in [74a71b2](https://github.com/brave/brave-ios/pull/7153/commits/74a71b2c6cf261554fe1764e5a653ce8752d20ff).